middleware/reverse/setup.go

```diff
@@ -117,8 +121,12 @@ func reverseParse(c *caddy.Controller) (nets networks, fall bool, err error) {
 				if ipnet.IP.To4() == nil {
 					regexIP = regexMatchV6
 				}
+				prefix := "^"
+				if wildcard {
+					prefix += ".*"
```

Comment:
That doesn't make sense. 

~~~
diff --git a/middleware/reverse/setup.go b/middleware/reverse/setup.go
index ee426f85..663ebea6 100644
--- a/middleware/reverse/setup.go
+++ b/middleware/reverse/setup.go
@@ -121,7 +121,7 @@ func reverseParse(c *caddy.Controller) (nets networks, fall bool, err error) {
                                if ipnet.IP.To4() == nil {
                                        regexIP = regexMatchV6
                                }
-                               prefix := "^"
+                               prefix := ""
                                if wildcard {
                                        prefix += ".*"
                                }
diff --git a/middleware/reverse/setup_test.go b/middleware/reverse/setup_test.go
index 2198c1e3..65671a72 100644
--- a/middleware/reverse/setup_test.go
+++ b/middleware/reverse/setup_test.go
@@ -14,10 +14,10 @@ func TestSetupParse(t *testing.T) {
        _, net4, _ := net.ParseCIDR("10.1.1.0/24")
        _, net6, _ := net.ParseCIDR("fd01::/64")
 
-       regexIP6, _ := regexp.Compile("^ip-" + regexMatchV6 + "\\.domain\\.com\\.$")
-       regexIpv4dynamic, _ := regexp.Compile("^dynamic-" + regexMatchV4 + "-intern\\.dynamic\\.domain\\.com\\.$")
-       regexIpv6dynamic, _ := regexp.Compile("^dynamic-" + regexMatchV6 + "-intern\\.dynamic\\.domain\\.com\\.$")
-       regexIpv4vpndynamic, _ := regexp.Compile("^dynamic-" + regexMatchV4 + "-vpn\\.dynamic\\.domain\\.com\\.$")
+       regexIP6, _ := regexp.Compile("ip-" + regexMatchV6 + "\\.domain\\.com\\.$")
+       regexIpv4dynamic, _ := regexp.Compile("dynamic-" + regexMatchV4 + "-intern\\.dynamic\\.domain\\.com\\.$")
+       regexIpv6dynamic, _ := regexp.Compile("dynamic-" + regexMatchV6 + "-intern\\.dynamic\\.domain\\.com\\.$")
+       regexIpv4vpndynamic, _ := regexp.Compile("dynamic-" + regexMatchV4 + "-vpn\\.dynamic\\.domain\\.com\\.$")
 
        serverBlockKeys := []string{"domain.com.:8053", "dynamic.domain.com.:8053"}
 
~~~
fixes the tests